ClosedXML/Excel/Cells/XLCell.cs

```diff
@@ -1755,16 +1755,34 @@ private static bool TryGetDateTimeValue<T>(out T value, object currentValue)
                 return false;
             }
 
-            if (!DateTime.TryParse(currentValue.ToString(), out DateTime ts))
+            if (currentValue is T v) { value = v; return true; }
+
+            if (currentValue.IsNumber())
             {
-                value = default;
-                return false;
+                var dbl1 = Convert.ToDouble(currentValue);
+                if (dbl1.IsValidOADateNumber())
+                {
+                    value = (T)Convert.ChangeType(DateTime.FromOADate(dbl1), typeof(T));
+                    return true;
+            }
             }
 
+            if (DateTime.TryParse(currentValue.ToString(), out DateTime ts))
+            {
             value = (T)Convert.ChangeType(ts, typeof(T));
             return true;
         }
 
+            if (Double.TryParse(currentValue.ObjectToInvariantString(), out Double dbl2) && dbl2.IsValidOADateNumber())
+            {
+                value = (T)Convert.ChangeType(DateTime.FromOADate(dbl2), typeof(T));
+                return true;
+            }
```

Comment:
This block is not covered by tests; there might be something like this:
```
[Test]
public void TryGetValue_DateTime_Good3()
{
    IXLWorksheet ws = new XLWorkbook().Worksheets.Add("Sheet1");
    bool success = ws.Cell("A1").SetFormulaA1("=\"44\"&\"020\"").TryGetValue(out DateTime outValue);
    Assert.IsTrue(success);
    Assert.AreEqual(new DateTime(2020, 07, 08), outValue);
}
```
But I am not 100% sure this should work this way. Excel, for example, leaves text to be text, even if a cell type is Date

![image](https://user-images.githubusercontent.com/19576939/86889690-34a22a80-c10d-11ea-817d-5864f0ea9fb9.png)

Reply:
But this does work in Excel:
![image](https://user-images.githubusercontent.com/145854/87810906-05eb2900-c85e-11ea-900d-91224287151b.png)

We could possibly move the string to Date conversion into the calculation engine, but I don't really see a down-side to leaving it here. If the user calls `cell.GetValue<DateTime>()` then I assume he really wants to get out a DateTime as much as possible.

Reply:
After thinking about it over the weekend, I now agree that we shouldn't do a "double-hop" to coerce values from one type to another. Converting a string to a number and then a number to a date, would be a "double-hop". This is only for the `TryGetValue<>()` cases. In the calc engine, coercing can be a bit more lenient, as is already the case. See updated unit test: `TryGetValue_DateTime_BadButFormulaGood`